app/controllers/ForumTopic.scala

```diff
@@ -55,20 +57,27 @@ final class ForumTopic(env: Env) extends LilaController(env) with ForumControlle
         then notFound
         else
           for
-            unsub       <- ctx.me soUse env.timeline.status(s"forum:${topic.id}")
-            canRead     <- access.isGrantedRead(categ.slug)
+            unsub   <- ctx.me soUse env.timeline.status(s"forum:${topic.id}")
+            canRead <- access.isGrantedRead(categ.slug)
+            topicUserId = topic.userId.getOrElse(UserId(""))
+            relation <- ctx.userId.so(
+              env.relation.api.fetchRelation(topicUserId: UserId, _)
+            )
             canWrite    <- access.isGrantedWrite(categ.slug, tryingToPostAsMod = true)
             canModCateg <- access.isGrantedMod(categ.slug)
-            inOwnTeam   <- ~(categ.team, ctx.me).mapN(env.team.api.isLeader(_, _))
+            isUblog      = topic.ublogId.isDefined
+            replyBlocked = relation.exists(_ == Block) && !canModCateg && isUblog
+            inOwnTeam <- ~(categ.team, ctx.me).mapN(env.team.api.isLeader(_, _))
             form <- ctx.me
-              .filter(_ => canWrite && topic.open && !topic.isOld)
+              .filter(_ => canWrite && topic.open && !topic.isOld && !replyBlocked)
```

Comment:
all the above is copy-pasted from `ForumPost` so the same comments apply. Should probably be refactored away.

Reply:
I have made a function in forumaccess now.